runtime/src/iree/vm/list.c

```diff
@@ -163,10 +163,10 @@ IREE_API_EXPORT iree_status_t iree_vm_list_initialize(
   iree_host_size_t required_storage_size =
       storage_offset + iree_host_align(capacity * element_size, 8);
```

Comment:
(Commenting for threading)

I've also been testing this change with
```cmake
    "-Wno-int-to-void-pointer-cast"
    "-Wno-int-to-pointer-cast"
    "-Wno-pointer-to-int-cast"
```

since we have code that does this:

```c
static inline iree_host_size_t iree_host_align(iree_host_size_t value,
                                               iree_host_size_t alignment)
// ...
  contents->buffer.data = (void*)iree_host_align(
      (uintptr_t)contents + sizeof(*contents), IREE_FILE_BASE_ALIGNMENT);
```

That produces errors like this when `iree_host_size_t` != `size_t`:

```
iree/base/internal/file_io.c:141:27: error: cast to 'void *' from smaller integer type 'iree_host_size_t' (aka 'unsigned int') [-Werror,-Wint-to-void-pointer-cast]
[build]   contents->buffer.data = (void*)iree_host_align(
[build]                           ^~~~~~~~~~~~~~~~~~~~~~~
```

---

Do we want to address that somehow? Being explicit about types is nice, but I don't want to bend C++ too much :P